arcgis-ios-sdk-samples/Layers/Group layers/LayersTableViewController.swift

```diff
@@ -35,23 +44,52 @@ class LayersTableViewController: UITableViewController, GroupLayersCellDelegate,
     }
     
     override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        let cell = tableView.dequeueReusableCell(withIdentifier: "cell", for: indexPath) as! GroupLayersCell
-        
-        guard let groupLayer = layers[indexPath.section] as? AGSGroupLayer, let childLayers = groupLayer.layers as? [AGSLayer] else { return cell }
-        
-        let childLayer = childLayers[indexPath.row]
-        
-        // Set label.
-        cell.layerNameLabel.text = formattedValue(of: childLayer.name)
-        
-        // Set state of the switch.
-        cell.layerVisibilitySwitch.isOn = childLayer.isVisible
-        cell.layerVisibilitySwitch.isEnabled = groupLayer.isVisible
-        
-        // To update the visibility of operational layers on switch toggle.
-        cell.delegate = self
-        
-        return cell
+        guard let groupLayer = layers[indexPath.section] as? AGSGroupLayer else { fatalError("Unknown cell type") }
+        switch groupLayer.visibilityMode {
+        case .independent:
+            let cell = tableView.dequeueReusableCell(withIdentifier: "switchCell", for: indexPath) as! GroupLayersCell
+            guard let childLayers = groupLayer.layers as? [AGSLayer] else { return cell }
+            let childLayer = childLayers[indexPath.row]
+            
+            // Set label.
+            cell.layerNameLabel.text = formattedValue(of: childLayer.name)
+            
+            // Set state of the cell and switch.
+            cell.isUserInteractionEnabled = groupLayer.isVisible
+            cell.layerVisibilitySwitch.isOn = childLayer.isVisible
+            cell.layerVisibilitySwitch.isEnabled = groupLayer.isVisible
+            
+            // To update the visibility of operational layers on switch toggle.
+            cell.delegate = self
+            
+            return cell
+        case .exclusive:
+            let cell = tableView.dequeueReusableCell(withIdentifier: "exclusiveCell", for: indexPath)
+            guard let childLayers = groupLayer.layers as? [AGSLayer] else { return cell }
+            
+            let childLayer = childLayers[indexPath.row]
+            
+            // Set label.
+            cell.textLabel?.text = formattedValue(of: childLayer.name)
+            // Enable or disable the cell accordingly.
+            cell.isUserInteractionEnabled = groupLayer.isVisible
+            cell.textLabel?.isEnabled = groupLayer.isVisible
+            // Adjust the tint if the cell is enabled.
+            if groupLayer.isVisible {
+                cell.tintAdjustmentMode = .automatic
+            } else {
+                cell.tintAdjustmentMode = .dimmed
+            }
```

Comment:
This is correct, but you still need to set the tint color of the cell to the tint color of the view. Setting the `tintAdjustmentMode` means that you don't need to set a different tint color depending on whether the group layer is visible. With the adjustment mode as dimmed, all controls in the cell will have the correct appearance when disabled.